core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -74,13 +74,12 @@ class SparkHadoopUtil extends Logging {
     }
   }
 
-  /**
-   * Return an appropriate (subclass) of Configuration. Creating config can initializes some Hadoop
-   * subsystems.
-   */
-  def newConfiguration(conf: SparkConf): Configuration = {
-    val hadoopConf = new Configuration()
 
+  /**
+    * Appends S3-specific, spark.hadoop.*, and spark.spark.buffer.size configurations to a Hadoop
```

Comment:
Extra "spark." here in "spark.spark.buffer.size".
This might be a dumb question, but it feels kind of funny that we set these things when a `Configuration` is created everywhere except one place. Is it because the `HiveConf` necessarily comes from somewhere else that Spark isn't initializing? Just trying to rationalize treating it specially in this one case